server/channels/api4/job_test.go

```diff
@@ -261,20 +279,26 @@ func TestDownloadJob(t *testing.T) {
 	// as a system admin, we should get a not found status.
 	_, err = th.App.Srv().Store().Job().Save(job)
 	require.NoError(t, err)
-	defer th.App.Srv().Store().Job().Delete(job.Id)
+	defer func() {
+		if _, delErr := th.App.Srv().Store().Job().Delete(job.Id); delErr != nil {
+			t.Logf("Failed to delete job %s: %v", job.Id, delErr)
+		}
+	}()
 
 	filePath := "./data/export/" + job.Id + "/testdat.txt"
 	mkdirAllErr := os.MkdirAll(filepath.Dir(filePath), 0770)
 	require.NoError(t, mkdirAllErr)
-	os.Create(filePath)
+
+	_, createErr := os.Create(filePath)
+	require.NoError(t, createErr)
 
 	// Normal user cannot download the results of these job (not the right permission)
 	_, resp, err = th.Client.DownloadJob(context.Background(), job.Id)
 	require.Error(t, err)
 	CheckForbiddenStatus(t, resp)
-	th.SystemManagerClient.DownloadJob(context.Background(), job.Id)
-	// System manager with default permissions cannot download the results of these job (Doesn't have correct permissions)
+
 	_, resp, err = th.SystemManagerClient.DownloadJob(context.Background(), job.Id)
+	// System manager with default permissions cannot download the results of these job (Doesn't have correct permissions)
```

Comment:
Unrelated change?

Reply:
Yes I messed upp a bit, now added the correct handling and removed the unwanted changes